src/Microsoft.AspNet.Identity/UserManager.cs

```diff
@@ -1063,10 +1062,9 @@ public virtual Task<IdentityResult> RemoveClaimAsync(TUser user, Claim claim)
                 throw new ArgumentNullException("roles");
             }
 
-            var userRoles = await userRoleStore.GetRolesAsync(user, CancellationToken);
             foreach (var role in roles)
```

Comment:
I still don't see what would prevent `await userManager.AddToRolesAsync(user, new [] {"admin", "admin"})` from trying to add "admin" twice, e.g. the LINQ queries above won't return UserRoles that haven't been saved to the database, unless we save for each iteration it won't help. But if you say it handles it, I am good :smile: